source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2386,6 +2386,48 @@ lldb::addr_t SwiftLanguageRuntime::FixupAddress(lldb::addr_t addr,
   return addr;
 }
 
+const swift::reflection::TypeInfo *
+SwiftLanguageRuntime::GetTypeInfo(CompilerType type) {
+  swift::CanType swift_can_type(GetCanonicalSwiftType(type));
+  ConstString mangled_name(type.GetMangledTypeName());
+  StringRef mangled_no_prefix =
+      swift::Demangle::dropSwiftManglingPrefix(mangled_name.GetStringRef());
+  swift::Demangle::Demangler Dem;
+  auto demangled = Dem.demangleType(mangled_no_prefix);
+  auto *type_ref = swift::Demangle::decodeMangledType(
+      reflection_ctx->getBuilder(), demangled);
+  if (!type_ref)
+    return nullptr;
+  return reflection_ctx->getBuilder().getTypeConverter().getTypeInfo(type_ref);
+}
+
+bool SwiftLanguageRuntime::IsIndirectResilientType(CompilerType type) {
```

Comment:
Suggestion for a better name: `IsStoredInlineInBuffer()`

Also I believe this does the same thing as `SwiftASTContext::GetAllocationStrategy()`, except it uses RemoteAST and not IRGen. Should the latter be phased out in favor of the former (same with other IRGen TypeInfo-based queries)?

Reply:
In particular, your new implementation will never give an answer of `TypeAllocationStrategy::eDynamic` because RemoteMirrors can always give a definite result using runtime metadata.